Kernel/VirtIO/VirtIORNG.cpp

```diff
@@ -23,7 +23,12 @@ VirtIORNG::VirtIORNG(PCI::Address address)
         m_entropy_buffer = MM.allocate_contiguous_kernel_region(PAGE_SIZE, "VirtIORNG", Region::Access::Read | Region::Access::Write);
         if (m_entropy_buffer) {
             memset(m_entropy_buffer->vaddr().as_ptr(), 0, m_entropy_buffer->size());
-            supply_buffer_and_notify(REQUESTQ, ScatterGatherList::create_from_physical(m_entropy_buffer->physical_page(0)->paddr(), m_entropy_buffer->size()), BufferType::DeviceWritable, m_entropy_buffer->vaddr().as_ptr());
+            auto &queue = get_queue(REQUESTQ);
+            ScopedSpinLock lock{queue.lock()};
+            VirtIOQueueChain chain{queue};
+            chain.add_buffer_to_chain(m_entropy_buffer->physical_page(0)->paddr().get(), m_entropy_buffer->size(), BufferType::DeviceWritable);
+            chain.commit();
+            notify_queue(REQUESTQ);
```

Comment:
should the device be responsible for notifiying the underlying queue? this seems a bit flimsy (this also doesnt check for the case in which the device explicitly disabled interrupts).

Reply:
Yeah, I was considering making this a step in `chain.commit()`, but that would require the chain to know which virtio device it's attached to + which virtio queue index the underlying queue uses.

Reply:
Really, only the device has the information needed to alert the queue and know whether we should alert the queue or not.